hpat/tests/test_io.py

```diff
@@ -363,49 +364,67 @@ def test_impl():
 
-    @unittest.skipIf(platform.system() == 'Windows', "error on windows")
     def test_csv1(self):
-        def test_impl():
-            return pd.read_csv("csv_data1.csv",
-                               names=['A', 'B', 'C', 'D'],
-                               dtype={'A': np.int, 'B': np.float, 'C': np.float, 'D': np.int},
-                               )
+        # TODO: w/a for Numba issue with int typing rules infering intp for integers literals
+        # unlike NumPy which uses int32 by default - causes dtype mismatch on Windows 64 bit 
+        if platform.system() == 'Windows' and not IS_32BITS:
+            def test_impl():
+                return pd.read_csv("csv_data1.csv",
+                                   names=['A', 'B', 'C', 'D'],
+                                   dtype={'A': np.intp, 'B': np.float, 'C': np.float, 'D': np.intp},
+                                   )
+        else:
+            def test_impl():
+                return pd.read_csv("csv_data1.csv",
+                                   names=['A', 'B', 'C', 'D'],
+                                   dtype={'A': np.int, 'B': np.float, 'C': np.float, 'D': np.int},
+                                   )
         hpat_func = hpat.jit(test_impl)
         pd.testing.assert_frame_equal(hpat_func(), test_impl())
 
-    @unittest.skip('Error - fix needed\n'
-                   'NUMA_PES=3 build')
     def test_csv_keys1(self):
-        def test_impl():
-            dtype = {'A': np.int, 'B': np.float, 'C': np.float, 'D': np.int}
-            return pd.read_csv("csv_data1.csv",
-                               names=dtype.keys(),
-                               dtype=dtype,
-                               )
+        if platform.system() == 'Windows' and not IS_32BITS:
+            def test_impl():
+                dtype = {'A': np.intp, 'B': np.float, 'C': np.float, 'D': np.intp}
+                return pd.read_csv("csv_data1.csv",
+                                   names=dtype.keys(),
+                                   dtype=dtype,
+                                   )
+        else:
+            def test_impl():
+                dtype = {'A': np.int, 'B': np.float, 'C': np.float, 'D': np.int}
+                return pd.read_csv("csv_data1.csv",
+                                   names=dtype.keys(),
+                                   dtype=dtype,
+                                   )
         hpat_func = hpat.jit(test_impl)
         pd.testing.assert_frame_equal(hpat_func(), test_impl())
 
-    @unittest.skipIf(platform.system() == 'Windows', "error on windows")
     def test_csv_const_dtype1(self):
-        def test_impl():
-            dtype = {'A': 'int', 'B': 'float64', 'C': 'float', 'D': 'int64'}
-            return pd.read_csv("csv_data1.csv",
-                               names=dtype.keys(),
-                               dtype=dtype,
-                               )
+        if platform.system() == 'Windows' and not IS_32BITS:
```

Comment:
On Linux 64 bit, what type will be used for `A`?

Reply:
The expectation is to have default np.int for A which is np.int64 on both Linux and MacOS (both 64-bit OS). 
For Windows 32-bit we can still use old test_impl, as it's not affected by the issue, and we'll have both A's dtype and Numba's types.intp int32. And only for Windows 64-bit we will have these 'tweaked' tests, which just fail otherwise.

Reply:
I have an idea.
Copy this original test (and others rlevent to this issue) to https://github.com/IntelPython/hpat/blob/master/hpat/tests/test_hpat_jit.py module to collect such issues.
Modify this test as you proposed before (with explicit typing).
Name of copied test must be the same as this one with `_issue` suffix to find original test later. Or add comment in the test body where it was copied from.